gateway-server/src/main/java/org/apache/knox/gateway/webshell/WebShellTokenUtils.java

```diff
@@ -0,0 +1,247 @@
+package org.apache.knox.gateway.webshell;
+
+import com.nimbusds.jose.JWSHeader;
+import org.apache.knox.gateway.config.GatewayConfig;
+import org.apache.knox.gateway.i18n.messages.MessagesFactory;
+//todo: add gateway-provider-security-jwt dependency?
+import org.apache.knox.gateway.provider.federation.jwt.JWTMessages;
+import org.apache.knox.gateway.provider.federation.jwt.filter.SignatureVerificationCache;
+import org.apache.knox.gateway.services.GatewayServices;
+import org.apache.knox.gateway.services.ServiceType;
+import org.apache.knox.gateway.services.security.token.JWTokenAuthority;
+import org.apache.knox.gateway.services.security.token.TokenMetadata;
+import org.apache.knox.gateway.services.security.token.TokenServiceException;
+import org.apache.knox.gateway.services.security.token.TokenStateService;
+import org.apache.knox.gateway.services.security.token.TokenUtils;
+import org.apache.knox.gateway.services.security.token.UnknownTokenException;
+import org.apache.knox.gateway.services.security.token.impl.JWT;
+import org.apache.knox.gateway.services.security.token.impl.JWTToken;
+import org.apache.knox.gateway.services.topology.TopologyService;
+import org.apache.knox.gateway.topology.Service;
+import org.apache.knox.gateway.topology.Topology;
+import org.apache.knox.gateway.util.CertificateUtils;
+import org.apache.knox.gateway.util.Tokens;
+import org.eclipse.jetty.websocket.servlet.ServletUpgradeRequest;
+import java.net.HttpCookie;
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+public class WebShellTokenUtils {
```

Comment:
We can make this simple by just relying on tokenstateservice. The assumption here is that Knox issues the token and put it in the tokenstateservice. All we need to do is check the token timestamp and expiry. We need to run this by @pzampino and @smolnar82